lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -532,10 +653,16 @@ func (c *ContainerProxy) Signal(vc *viccontainer.VicContainer, sig uint64) error
 			}
 		}
 
-		c.Wait(vc, -1*time.Second)
+		// Wait awhile to see if signal killed the container
+		if _, _, _, err := c.Wait(vc, killWaitBeforeForce); err != nil {
```

Comment:
We are waiting above before we check the container state again; why don't we just call stop in that code? Also, is calling stop a good idea? I am not sure if sending a KILL signal to the container and stop are equivalent from the client's perspective. 

In the type switch above we are also missing a default case.